index.bs

```diff
@@ -3927,19 +3933,40 @@ The [=remote end steps=] with |command parameters| are:
    1. Otherwise, set the |context|'s [=layout viewport=] to the
       implementation-defined default.
 
+1. Run the [[cssom-view-1#resizing-viewports]] steps.
+
 1. If |command parameters| [=map/contains=] the <code>devicePixelRatio</code> field:
 
    1. Let |device pixel ratio| be the |command
       parameters|["<code>devicePixelRatio</code>"].
 
-   1. If |device pixel ratio| is not null, change the size of the [=CSS Pixel=]
-      in |context|'s [=layout viewport=] such that it corresponds to |device
-      pixel ratio| in device pixels.
+   1. For the |context| and all [=descendant navigables=]:
 
-   1. Otherwise, set the size of the [=CSS Pixel=] in |context|'s [=layout
-      viewport=] to the implementation-defined default.
+      1. If |device pixel ratio| is not null:
 
-1. Run the [[cssom-view-1#resizing-viewports]] steps.
+         1. When the [=select an image source from a source set=] are run, act as if
+            the implementation's pixel density was set to |device pixel ratio| when selecting an image.
+
+         1. For the purposes of the [=resolution media feature=], act as if
+            the implementation's resolution is |device pixel ratio| dppx scaled by the page zoom.
+
+         1. When the [=determine the device pixel ratio=] steps are run, return |device pixel ratio|.
+
+         1. Run [=evaluate media queries and report changes=] for [=document=] currently loaded
+            in a specified |context|.
+
+      1. Otherwise:
+
+         1. When the [=select an image source from a source set=] steps are run, use the implementation's default behavior,
+            without any changes made by previous invocations of these steps.
+
+         1. For the purposes of the [=resolution media feature=], use the implementation's default behavior,
+            without any changes made by previous invocations of these steps.
+
+         1. When the [=determine the device pixel ratio=] steps are run, return the implemenation's default value.
```

Comment:
This seems to be yet another new algorithm; you actually want to say "run the original algorithm".

Maybe the right way to do this one is actually to define the algorithm we want in the "patches to other specifications" section, something like:

```
Insert the following steps at the start of the [=determine the device pixel ratio=] algorithm:

1. If [=device pixel ratio overrides=] [=map/contains=] |window|'s [=window/navigable=], return [=device pixel ratio overrides=][|window|=].
```

Then in this section you'd need to define `device pixel ratio overrides` as a weak map between navigables and the dpr override, and set / remove entries at the appropriate time.

Reply:
Alright, I've updated this part. Let me know how it looks